lib/generateClass.js

```diff
@@ -540,6 +545,8 @@ function writeFromFhirProfile(def, specs, fhir, fhirProfile, cw) {
   const allFhirElements = preProcessChoiceFields(fhirProfile.snapshot.element, specs, fhir);
   preProcessFieldMappings(allFhirElements, specs, def);
 
+  // Determine isQuantity outside of the iteration loop so it is more efficient.  See note below for reason we need isQuantity.
```

Comment:
I'm not sure exactly what you mean by "recognize this kind of situation more generically", but... consider this an interim approach for now.  I think where we want to go is to allow the components of `concept` (even though it is technically a primitive) to be mapped just like other things.  So instead of:
```
map Units to code
```
(which is what we have now and is problematic), we can have something more like this:
```
map Units[concept].code to code
map Units[concept].system to system
map Units[concept].display to unit
```

This will be more like everything else and should hopefully require very little special case code (just to deal, generally, with being able to map components of `concept`).